src/components/dialog/dialog.js

```diff
@@ -426,6 +426,7 @@ function MdDialogDirective($$rAF, $mdTheming, $mdDialog) {
  *   - `preserveScope` - `{boolean=}`: whether to preserve the scope when the element is removed. Default is false
  *   - `disableParentScroll` - `{boolean=}`: Whether to disable scrolling while the dialog is open.
  *     Default true.
+ *   - `beforeClose` - `{function=}`: A function that returns a promise. If the promise resolved, the dialog will close.
```

Comment:
There seems to be a typo with the `resolved` ->  `If the promise resolves, the dialog will close immediately`